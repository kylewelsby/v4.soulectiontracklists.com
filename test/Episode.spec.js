/**
 * @jest-environment node
 */
import nock from 'nock'
import {
  useSoundcloud,
  useFlatTracklist,
  useFormattedSummary,
} from '../pages/episodes/episode.js'

describe('useSoundcloud', () => {
  beforeEach(() => {
    nock('https://api-mobi.soundcloud.com')
      .get('/resolve')
      .query({
        permalink_url:
          'https://soundcloud.com/soulection/soulection-radio-show-478',
        client_id: 'iZIs9mchVcX5lhVRyQGGAYlNPVldzAoX',
      })
      .replyWithFile(200, `${__dirname}/__fixtures__/soundcloud-resolve.json`)
  })
  test('returns soundcloud data', async () => {
    const soundcloud = await useSoundcloud(
      'https://soundcloud.com/soulection/soulection-radio-show-478'
    )
    expect(soundcloud).toMatchSnapshot()
  })
})

describe('useFlatTracklist', () => {
  test('returns flatterned tracklist', async () => {
    nock.recorder.rec()
    const tracklist = await useFlatTracklist(
      {
        $content: jest.fn((path, slug) => {
          let res = jest.fn().mockResolvedValue({
            title: 'Bad',
            artwork: 'images/michael-jackson/tracks/bad.jpg',
          })
          if (path === 'artists') {
            res = jest.fn().mockResolvedValue([
              {
                title: 'Michael Jackson',
              },
            ])
          }
          return {
            limit: jest.fn().mockReturnThis(),
            fetch: res,
          }
        }),
      },
      {
        artwork: 'images/episodes/478.jpg',
        title: 'Show #478',
        sessions: [
          {
            name: "Joe Kay's Set",
            tracks: ['before - after'],
            refs: ['artists/afta-1/tracks/tvttoo-001-sit-still.md'],
            cue: ['00:03:00'],
          },
        ],
      }
    )
    expect(tracklist).toMatchSnapshot()
  })
})

describe('useFormattedSummary', () => {
  test('does thing', () => {
    const $t = jest.fn().mockImplementation((key, options) => {
      if (key === 'episode.soloSession') {
        return `${options.tracks} tracks`
      } else if (key === 'episode.soloSessionWithInterview') {
        return `${options.tracks} tracks with Interview`
      } else if (key === 'episode.multiSession') {
        return `${options.tracks} tracks across ${options.sessions} sets`
      } else if (key === 'episode.multiSessionWithInterview') {
        return `${options.tracks} tracks across ${options.sessions} sets with Interview`
      } else {
        throw new Error(`unhandled tag ${key}`)
      }
    })
    const soloSummary = useFormattedSummary({ $t }, [
      {
        name: "Joe Kay's Set",
        tracks: ['hello - world'],
      },
    ])
    expect(soloSummary).toMatchSnapshot()

    const multiSessionWithInterview = useFormattedSummary({ $t }, [
      {
        name: "Joe Kay's Set",
        tracks: ['hello - world'],
      },
      {
        name: 'Interview with Set',
        tracks: ['hello - world'],
      },
    ])
    expect(multiSessionWithInterview).toMatchSnapshot()

    const soloSessionWithInterview = useFormattedSummary({ $t }, [
      {
        name: 'Interview with Set',
        tracks: ['hello - world'],
      },
    ])
    expect(soloSessionWithInterview).toMatchSnapshot()

    const multiSession = useFormattedSummary({ $t }, [
      {
        name: '1 Set',
        tracks: ['hello - world'],
      },
      {
        name: '2 Set',
        tracks: ['hello - world'],
      },
    ])
    expect(multiSession).toMatchSnapshot()
  })
})
