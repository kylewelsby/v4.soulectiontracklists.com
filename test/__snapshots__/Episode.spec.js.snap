// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFlatTracklist returns flatterned tracklist 1`] = `
Array [
  Object {
    "artist": "Michael Jackson",
    "artwork": "https://firebase.soulectiontracklists.com/cdn/image/images/michael-jackson/tracks/bad.jpg",
    "episodeArtwork": "https://firebase.soulectiontracklists.com/cdn/image/images/episodes/478.jpg",
    "episodeTitle": "Show #478",
    "position": 180,
    "sessionTitle": "Joe Kay's Set",
    "title": "Bad",
  },
]
`;

exports[`useFormattedSummary does thing 1`] = `"1 tracks"`;

exports[`useFormattedSummary does thing 2`] = `"2 tracks across 2 sets with Interview"`;

exports[`useFormattedSummary does thing 3`] = `"1 tracks with Interview"`;

exports[`useFormattedSummary does thing 4`] = `"2 tracks across 2 sets"`;

exports[`useSoundcloud returns soundcloud data 1`] = `
Object {
  "artwork_url": "https://i1.sndcdn.com/artworks-8PcfHcO5Me9uFNB6-QxGyyQ-large.png",
  "caption": null,
  "comment_count": 125,
  "commentable": true,
  "created_at": "2020-10-22T18:32:05Z",
  "description": "▶ Soulection ~ The Sound of Tomorrow
▶ instagram.com/soulection | twitter.com/soulection
▶ Show #478 Track List: soulectiontracklists.com/episodes/478
▶ smarturl.it/soul478
▶ Discover: @soulection @joekay
- The SoundCloud uploads are Explicit
- The Clean Version is up on Apple Music 1
** Soulection Radio Shows on Apple Music 1 are archived here on SoundCloud 6 days after they air live on Saturday **
**** Full track Lists are now available soulectiontracklists.com ****
→ If you are interested in hearing all of our recent shows go here: soundcloud.com/soulection/sets/soulection-radio-sessions
→ Stay tuned to our site for more info on tour dates, releases, and much more via soulection.com
→ Find us on Instagram to see what goes on behind the scenes: instagram.com/soulection
Design by @Sangobeats
→ We want you to get involved with the Soulection Radio Cover designs. If you have a dope photo or design that you think fits us, find us on IG and hashtag: 'SoulectionRadio'",
  "display_date": "2020-10-23T13:00:26Z",
  "download_count": 809,
  "downloadable": true,
  "duration": 7257208,
  "embeddable_by": "all",
  "full_duration": 7257208,
  "genre": "",
  "has_downloads_left": true,
  "id": 916213124,
  "kind": "track",
  "label_name": null,
  "last_modified": "2020-11-10T19:44:54Z",
  "license": "all-rights-reserved",
  "likes_count": 1845,
  "media": Object {
    "transcodings": Array [
      Object {
        "duration": 7257208,
        "format": Object {
          "mime_type": "audio/mpeg",
          "protocol": "hls",
        },
        "preset": "mp3_0_1",
        "quality": "sq",
        "snipped": false,
        "url": "https://api-mobi.soundcloud.com/media/soundcloud:tracks:916213124/a9098779-5794-46dc-b303-28c1c5bb6f57/stream/hls",
      },
      Object {
        "duration": 7257208,
        "format": Object {
          "mime_type": "audio/mpeg",
          "protocol": "progressive",
        },
        "preset": "mp3_0_1",
        "quality": "sq",
        "snipped": false,
        "url": "https://api-mobi.soundcloud.com/media/soundcloud:tracks:916213124/a9098779-5794-46dc-b303-28c1c5bb6f57/stream/progressive",
      },
    ],
  },
  "monetization_model": "NOT_APPLICABLE",
  "permalink": "soulection-radio-show-478",
  "permalink_url": "https://soundcloud.com/soulection/soulection-radio-show-478",
  "playback_count": 46828,
  "policy": "ALLOW",
  "public": true,
  "publisher_metadata": Object {
    "album_title": "17 October 2020",
    "artist": "Soulection Radio",
    "contains_music": true,
    "id": 916213124,
    "urn": "soundcloud:tracks:916213124",
  },
  "purchase_title": null,
  "purchase_url": null,
  "release_date": null,
  "reposts_count": 257,
  "secret_token": null,
  "sharing": "public",
  "state": "finished",
  "streamable": true,
  "tag_list": "SOULECTION",
  "title": "Soulection Radio Show #478",
  "uri": "https://api.soundcloud.com/tracks/916213124",
  "urn": "soundcloud:tracks:916213124",
  "user": Object {
    "avatar_url": "https://i1.sndcdn.com/avatars-IzJb4WdtZfIYTSrf-S5s4fA-large.jpg",
    "badges": Object {
      "pro_unlimited": true,
      "verified": false,
    },
    "city": "Los Angeles",
    "comments_count": 288,
    "country_code": "US",
    "created_at": "2009-12-30T21:22:00Z",
    "creator_subscription": Object {
      "product": Object {
        "id": "creator-pro-unlimited",
      },
    },
    "creator_subscriptions": Array [
      Object {
        "product": Object {
          "id": "creator-pro-unlimited",
        },
      },
    ],
    "description": "The Sound of Tomorrow

Discover Soulection: http://soulection.com

Soulection Radio •  Saturdays from 6-8pm (Pacific) on Apple Music 1  - Hosted by @JoeKay",
    "first_name": "",
    "followers_count": 519764,
    "followings_count": 51,
    "full_name": "",
    "groups_count": 0,
    "id": 483960,
    "kind": "user",
    "last_modified": "2020-11-06T16:07:02Z",
    "last_name": "",
    "likes_count": 1201,
    "permalink": "soulection",
    "permalink_url": "https://soundcloud.com/soulection",
    "playlist_count": 48,
    "playlist_likes_count": 43,
    "reposts_count": null,
    "track_count": 608,
    "uri": "https://api.soundcloud.com/users/483960",
    "urn": "soundcloud:users:483960",
    "username": "SOULECTION",
    "verified": false,
    "visuals": Object {
      "enabled": true,
      "tracking": null,
      "urn": "soundcloud:users:483960",
      "visuals": Array [
        Object {
          "entry_time": 0,
          "urn": "soundcloud:visuals:111888204",
          "visual_url": "https://i1.sndcdn.com/visuals-000000483960-yygjpN-original.jpg",
        },
      ],
    },
  },
  "user_id": 483960,
  "visuals": null,
  "waveform_url": "https://wave.sndcdn.com/vm0B359i6WEW_m.json",
}
`;
